src/traces/funnel/plot.js

```diff
@@ -0,0 +1,154 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var d3 = require('d3');
+var Lib = require('../../lib');
+var Drawing = require('../../components/drawing');
+var barPlot = require('../bar/plot');
+
+module.exports = function plot(gd, plotinfo, cdModule, traceLayer) {
+    barPlot(gd, plotinfo, cdModule, traceLayer);
+    plotConnectorRegions(gd, plotinfo, cdModule, traceLayer);
+    plotConnectorLines(gd, plotinfo, cdModule, traceLayer);
+};
+
+function plotConnectorRegions(gd, plotinfo, cdModule, traceLayer) {
+    var xa = plotinfo.xaxis;
+    var ya = plotinfo.yaxis;
+
+    Lib.makeTraceGroups(traceLayer, cdModule, 'trace bars').each(function(cd) {
+        var plotGroup = d3.select(this);
+        var cd0 = cd[0];
+        var trace = cd0.trace;
+
+        var group = Lib.ensureSingle(plotGroup, 'g', 'regions');
+
+        if(!trace.connector || !trace.connector.visible) {
+            group.remove();
+            return;
+        }
+
+        var isHorizontal = (trace.orientation === 'h');
+
+        if(!plotinfo.isRangePlot) cd0.node3 = plotGroup;
```

Comment:
We shouldn't need this line here. `node3` should be already linked up in `barPlot`.

Reply:
Good catch again! Fixed for waterfalls and funnels in https://github.com/plotly/plotly.js/pull/3817/commits/3d5b8ff69b7eb72fe6e5779d15cdaa3ed010412d.

Reply:
One more here:

https://github.com/plotly/plotly.js/blob/3d5b8ff69b7eb72fe6e5779d15cdaa3ed010412d/src/traces/funnel/plot.js#L92

Reply:
Good eye!
Fixed in https://github.com/plotly/plotly.js/pull/3817/commits/bdbbeb1b0438c25bcdf2566b7c19a2b64e2db745.

Reply:
Resolved.